thentos-core/src/Thentos/Backend/Core.hs

```diff
@@ -114,52 +117,58 @@ actionErrorToServantErr e = do
 
 
 class ThentosErrorToServantErr db where
-    thentosErrorToServantErr :: ThentosError db -> (Maybe (Priority, String), ServantErr)
+    -- | Convert a 'ThentosError' into a 'ServantErr'. If the first argument is 'Just' provided,
+    -- it must be invoked to create the returned error, otherwise a suitable default implementation
+    -- must be used for that purpose. This allows derived db's to adapt the details of the error
+    -- actually returned.
+    thentosErrorToServantErr :: Maybe MkServantErrFun -> ThentosError db
```

Comment:
> how else should the A3-specific implementation ensure that all errors have the A3-specific format

If this is what we're concerned with, I'd just opaquely newtype-wrap the error, and have a smart constructor (which would also fix the problem that right now `ServantErr`s can have their body and headers overwritten by `mkServantErr`, and nothing in the types indicates one should be worried about that). Then the types would guarantee the invariant we're concerned with is fulfilled. But in this case it seems to me like only the A3 DB type can be used in A3, so I'd really just apply the transformation once in the line I mentioned and be done with it. I still don't see how that's a weaker guarantee. 

More generally, if I have an `a :: Int`, and often apply a function `fn :: Int -> SomeType -> Int` to it, I wouldn't consider changing the type of `a` to `a :: (Int -> SomeType -> Int) -> Int`. It's more confusing and complicated, and there's nothing I can do with that that I couldn't do already.